From f576e7e2913d39c013a61793e8daf9943340cb25 Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Thu, 18 Sep 2014 14:19:52 -0400
Subject: [fs] nfsv4: Drop cast

Message-id: <1411050057-12417-30-git-send-email-steved@redhat.com>
Patchwork-id: 96219
O-Subject: [RHEL7.1 PATCH 29/94] NFSv4: Drop cast
Bugzilla: 1111708
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>

From: Himangi Saraogi <himangi774@gmail.com>

This patch does away with the cast on void * as it is unnecessary.

The following Coccinelle semantic patch was used for making the change:

@r@
expression x;
void* e;
type T;
identifier f;
@@

(
  *((T *)e)
|
  ((T *)x)[...]
|
  ((T *)x)->f
|
- (T *)
  e
)

Signed-off-by: Himangi Saraogi <himangi774@gmail.com>
Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Upstream Id: 8ee2b78a444c6715e873486794b5c4af705cd975
Bugzilla: 1111708
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfs/nfs4xdr.c b/fs/nfs/nfs4xdr.c
index 939ae60..e13b59d 100644
--- a/fs/nfs/nfs4xdr.c
+++ b/fs/nfs/nfs4xdr.c
@@ -7092,7 +7092,7 @@ static int nfs4_xdr_dec_reclaim_complete(struct rpc_rqst *rqstp,
  if (!status)
   status = decode_sequence(xdr, &res->seq_res, rqstp);
  if (!status)
-  status = decode_reclaim_complete(xdr, (void *)NULL);
+  status = decode_reclaim_complete(xdr, NULL);
  return status;
 }
 
-- 
1.7.1